roles/backend_setup/tasks/thin_pool_create.yml

```diff
@@ -22,15 +22,135 @@
+  register: thinpool_attrs
+  with_items: "{{ gluster_infra_thinpools }}"
+  loop_control:
+   index_var: index
+  when: > 
+   gluster_infra_thinpools is defined and gluster_infra_thinpools is not none and gluster_infra_thinpools|length >0 
+   and item is defined and item is not none and item.thinpoolname is defined and item.thinpoolname is defined  and item.vgname is defined
+   and lvtp_device_exists.results[index].stdout_lines is defined and "0" not in lvtp_device_exists.results[index].stdout_lines
+   and ((item.opts is defined and "raid" in item.opts) or item.meta_pvs is defined or item.meta_opts is defined)
+
+
+# https://github.com/ansible/ansible/issues/13262
+# block of tasks, which is executed when there are valid pools and pool items are configured.
+# also it checks if the pool doesn't already exists
+- name: Create a LV thinpool-data
+  lvol:
+     state: present
+     shrink: false
+     vg: "{{ item.vgname }}"
+     lv: "{{ item.thinpoolname }}"
+     pvs: "{{ item.pvs | default() }}"     
+     size: "{{ item.thinpoolsize | default('100%FREE') }}"
+     opts: " 
+             {{ item.opts | default('') }} "
+  with_items: "{{ gluster_infra_thinpools }}"
+  loop_control:
+   index_var: index
+  when: > 
+   gluster_infra_thinpools is defined and gluster_infra_thinpools is not none and gluster_infra_thinpools|length >0 
+   and item is defined and item is not none and item.thinpoolname is defined and item.thinpoolname is defined  and item.vgname is defined   
+   and lvtp_device_exists.results[index].stdout_lines is defined and "0" not in lvtp_device_exists.results[index].stdout_lines
+   and (thinpool_attrs.results[index].stdout is not defined or thinpool_attrs.results[index].stdout.find("t") != 0)
+   and ((item.opts is defined and "raid" in item.opts) or item.meta_pvs is defined or item.meta_opts is defined)
+
+- name: Make sure meta pv's exists in volume group
+  lvg:
+     state: present
+     vg: "{{ item.vgname }}"
+     pvs: "{{ item.meta_pvs }}"
+     pv_options: "--dataalignment 256K"
+  with_items: "{{ gluster_infra_thinpools }}"
+  loop_control:
+   index_var: index
+  when: > 
+   gluster_infra_thinpools is defined and gluster_infra_thinpools is not none and gluster_infra_thinpools|length >0 
+   and item is defined and item is not none and item.thinpoolname is defined and item.thinpoolname is defined  and item.vgname is defined   
+   and lvtp_device_exists.results[index].stdout_lines is defined and "0" not in lvtp_device_exists.results[index].stdout_lines
+   and (thinpool_attrs.results[index].stdout is not defined or thinpool_attrs.results[index].stdout.find("t") != 0)
+   and item.meta_pvs is defined
+
+- name: Create a LV thinpool-meta
+  lvol:
+     state: present
+     shrink: false
+     vg: "{{ item.vgname }}"
+     lv: "{{ item.thinpoolname }}_meta"
+     pvs: "{{ ((item.meta_pvs is defined and item.meta_pvs) or item.pvs) | default() }}"     
+     size: "{{ item.poolmetadatasize | default('4G') }}"
```

Comment:
@olafbuitelaar can you please make this 16G?
Ref: https://access.redhat.com/documentation/en-us/red_hat_gluster_storage/3.4/html-single/administration_guide/index#Brick_Configuration

I had a different patch to address this. I'll rebase and resolve the conflicts once this is merged.

Reply:
Done